lm_proposer.py

```diff
@@ -0,0 +1,218 @@
+
+from transformers import AutoTokenizer
+import nltk
+import openai
+import pickle as pkl
+import random
+import numpy as np
+from copy import deepcopy
+import os
+from typing import Dict, List
+import time
+
+openai.api_key = os.environ['openai_key']
```

Comment:
Nitpick:
Note that environment variables are typically written in upper case, so this would be `OPENAI_KEY`.